src/main/query/type.c

```diff
@@ -342,9 +338,9 @@ AerospikeQuery *AerospikeQuery_New(AerospikeClient *client, PyObject *args,
         Py_INCREF(client);
         return self;
     }
-    else {
-        return NULL;
-    }
+
+    AerospikeQuery_Type.tp_free(self);
+    return NULL;
```

Comment:
See my comment about the Client new function. Maybe the same object calling can be done here instead.

Reply:
aerospike.Query instances are created through [aerospike.Client.query()](https://aerospike-python-client.readthedocs.io/en/latest/client.html#aerospike.Client.query), so it might not be easy to do.